front/src/applications/editor/tools/rangeEdition/components/RangeEditionTrackRangeList.tsx

```diff
@@ -30,48 +33,43 @@ const TrackRangesList = () => {
   const ranges = entity.properties.track_ranges || [];
   const displayedRanges = showAll ? ranges : ranges.slice(0, DEFAULT_DISPLAYED_RANGES_COUNT);
 
-  const rangesByRoute = groupBy(displayedRanges, (range) => {
-    const isCurrentRange = compareTrackRange(range);
-    // eslint-disable-next-line no-restricted-syntax
-    for (const [routeKey, { trackRanges }] of Object.entries(routeElements)) {
-      if (isCurrentRange(routeExtra[routeKey]) || trackRanges.some(isCurrentRange)) {
-        return routeKey;
-      }
+  function makeFullTrackList() {
```

Comment:
This function is not very straightforward and makes the code difficult to read.

I think it would be clearer if you extract the code concerning the speedRestriction in another component. Then in the JSX, you can simply do this: 
```
      {displayedRanges.length === 0 ? (
        <p className="text-muted mt-3 text-center">
          {t('Editor.tools.range-edition.empty-linked-track-section')}
        </p>
      ) : (
        <ul className="list-unstyled">
          {!withRouteName ? (
            displayedRanges.map((tr, trackIndex) => (
              <TrackRange
                range={tr}
                index={trackIndex}
                speedRestrictionTool={speedRestrictionTool}
                key={`track-range-${tr.track}-${tr.begin}-${tr.end}-${tr.applicable_directions}-${trackIndex}`}
              />
            ))
          ) : (
            <SpeedRestrictonTrackRangeList displayedRanges={displayedRanges} />
          )}
        </ul>
      )}
```

Reply:
made some improvement